ostelco-ios-client/src/features/splash/SplashViewController.swift

```diff
@@ -26,208 +26,155 @@ class SplashViewController: UIViewController {
+        return false
+    }
+
+    func getContextFromCredentials(_ credentials: Credentials?) -> Promise<Context?> {
+        if setCredentials(credentials) {
+            print("setCredentials, calling getContext")
+            return self.getContext()
+        } else {
             performSegue(withIdentifier: "showLogin", sender: self)
-            return
-        }
-        let apiManager = APIManager.sharedInstance
-        let userManager = UserManager.sharedInstance
-        if (userManager.authToken != accessToken && userManager.authToken != nil) {
-            apiManager.wipeResources()
-            UserManager.sharedInstance.clear()
+            // Stop further processing of the promise chain
+            let rejected: Promise<Context?> = getRejectedPromise()
+            return rejected
         }
+    }
 
-        if (userManager.authToken != accessToken) {
-            apiManager.authHeader = "Bearer \(accessToken)"
-            UserManager.sharedInstance.authToken = accessToken
-        }
-        let optContext = try! await(getContext())
-        guard let context = optContext else {
+    func handleContext(_ context: Context?) -> Promise<Void?> {
+        if let context = context {
+            print("handleContext")
+            UserManager.sharedInstance.user = context.customer
+            if let region = context.getRegion() {
+                OnBoardingManager.sharedInstance.region = region
+                let segueIdentifier = getSegueFromRegionResponse(region: region)
+                print("handleContext switch to segue \(segueIdentifier)")
+                performSegue(withIdentifier: segueIdentifier, sender: self)
+            } else {
+                print("handleContext switch to showCountry")
+                performSegue(withIdentifier: "showCountry", sender: self)
+            }
+        } else {
             // for 404 the context will be empty
+            print("handleContext empty context 404")
             performSegue(withIdentifier: "showSignUp", sender: self)
-            return
         }
-        UserManager.sharedInstance.user = context.customer
+        let resolved: Promise<Void?> = getFullfilledPromise()
+        return resolved
+    }
 
-        var segueIdentifier: String
-        guard let region = context.getRegion() else {
-            performSegue(withIdentifier: "showCountry", sender: self)
-            return
+    func getRejectedPromise<T>() -> Promise<T?> {
+        return Promise<T?> { seal in
+            let message = "Terminating promise chain, ignore"
+            seal.reject(createError(message))
+        }
+    }
+
+    func getFullfilledPromise<T>() -> Promise<T?> {
+        let empty:T? = nil
+        return Promise<T?> { seal in
+            seal.resolve(empty, nil)
         }
-        OnBoardingManager.sharedInstance.region = region
+    }
+
+    func getRejectedContext() -> Promise<Context?> {
```

Comment:
This function seems to be the same as as the `getContext` on line 55 and doesn't seem to be used in this file